ui/src/app/views/profile-registration-step-status/component.tsx

```diff
@@ -0,0 +1,98 @@
+import * as React from 'react';
+
+import { Button } from 'app/components/buttons';
+import { ClrIcon } from 'app/components/icons';
+import colors from 'app/styles/colors';
+import { reactStyles } from 'app/utils';
+
+const styles = reactStyles({
+  container: {
+    backgroundColor: '#fff',
+    color: '#262262',
+    flex: '0 0 420px',
+    borderRadius: 8,
+    marginBottom: '10px',
+    padding: 21
+  },
+  title: {
+    color: colors.purple[0],
+    width: 420,
+    fontSize: 16,
+    fontWeight: 600,
+    marginBottom: 6
+  },
+  body: {
+    display: 'flex'
+  },
+  buttonContainer: {
+    width: 'calc(50%)',
+    marginTop: 6
+  },
+  detailsContainer: {
+    width: 'calc(50%)',
+    marginLeft: 40
+  },
+  button: {
+    height: 38
+  },
+  successButton: {
+    backgroundColor: '#8BC990',
+    cursor: 'default'
+  }
+});
+
+interface Props {
+  title: string;
+  wasBypassed: boolean;
+  isComplete: boolean;
+  incompleteButtonText: string;
+  completedButtonText: string;
+  completeStep: Function;
+}
+
+class ProfileRegistrationStepStatus extends React.Component<Props> {
```

Comment:
if this component doesn't have state, you could make it a React.FunctionComponent (see the `WorkspaceCardMenu` in workspace-list for example).  Then I would move this code into the profile component file, since it's not really a separate "view" and is unique to the profile

Reply:
Sounds good. I made the change to FunctionComponent. However, I think I prefer it in the separate file since it's easier to read shorter files.

Reply:
Discussed and decided that separate file is fine.